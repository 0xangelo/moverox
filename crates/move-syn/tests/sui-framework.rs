use move_syn::{File, sanitize_for_tokenizer};
use testresult::TestResult;
use unsynn::{IParse as _, ToTokens as _};

macro_rules! test_file {
    ($($file: ident),* $(,)?) => {$(
        #[test]
        fn $file() -> TestResult {
            let content = include_str!(concat!(
                "../../../move/sui-framework/sources/",
                stringify!($file),
                ".move"
            ));
            let ast: File = sanitize_for_tokenizer(&content).into_token_iter().parse_all()?;
            insta::assert_snapshot!(ast.tokens_to_string());
            Ok(())
        }
    )*};
}

test_file!(
    accumulator,
    address,
    authenticator_state,
    bag,
    balance,
    bcs,
    borrow,
    clock,
    coin,
    config,
    deny_list,
    display,
    dynamic_field,
    dynamic_object_field,
    event,
    hex,
    linked_table,
    math,
    object,
    object_bag,
    object_table,
    package,
    party,
    pay,
    priority_queue,
    prover,
    random,
    sui,
    table,
    table_vec,
    token,
    transfer,
    tx_context,
    types,
    url,
    vec_map,
    vec_set,
    versioned,
);
