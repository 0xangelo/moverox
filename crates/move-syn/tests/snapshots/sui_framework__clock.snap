---
source: crates/move-syn/tests/sui-framework.rs
expression: ast.tokens_to_string()
---
# [doc = " APIs for accessing time from move calls, via the `Clock`: a unique"] # [doc = " shared object that is created at 0x6 during genesis."] module sui :: clock ; # [doc = " Sender is not @0x0 the system address."] const ENotSystemAddress : u64 = 0 ; # [doc = " Singleton shared object that exposes time to Move calls.  This"] # [doc = " object is found at address 0x6, and can only be read (accessed"] # [doc = " via an immutable reference) by entry functions."] # [doc = ""] # [doc = " Entry Functions that attempt to accept `Clock` by mutable"] # [doc = " reference or value will fail to verify, and honest validators"] # [doc = " will not sign or execute transactions that use `Clock` as an"] # [doc = " input parameter, unless it is passed by immutable reference."] public struct Clock has key { id : UID , # [doc = " The clock's timestamp, which is set automatically by a"] # [doc = " system transaction every time consensus commits a"] # [doc = " schedule, or by `sui::clock::increment_for_testing` during"] # [doc = " testing."] timestamp_ms : u64 , } # [doc = " The `clock`'s current timestamp as a running total of"] # [doc = " milliseconds since an arbitrary point in the past."] public fun timestamp_ms (clock : & Clock) : u64 { clock . timestamp_ms } # [allow (unused_function)] # [doc = " Create and share the singleton Clock -- this function is"] # [doc = " called exactly once, during genesis."] fun create (ctx : & TxContext) { assert ! (ctx . sender () == @ 0x0 , ENotSystemAddress) ; transfer :: share_object (Clock { id : object :: clock () , timestamp_ms : 0 , }) } # [allow (unused_function)] fun consensus_commit_prologue (clock : & mut Clock , timestamp_ms : u64 , ctx : & TxContext) { assert ! (ctx . sender () == @ 0x0 , ENotSystemAddress) ; clock . timestamp_ms = timestamp_ms } # [test_only] # [doc = " Expose the functionality of `create()` (usually only done during"] # [doc = " genesis) for tests that want to create a Clock."] public fun create_for_testing (ctx : & mut TxContext) : Clock { Clock { id : object :: new (ctx) , timestamp_ms : 0 , } } # [test_only] # [doc = " For transactional tests (if a Clock is used as a shared object)."] public fun share_for_testing (clock : Clock) { transfer :: share_object (clock) } # [test_only] public fun increment_for_testing (clock : & mut Clock , tick : u64) { clock . timestamp_ms = clock . timestamp_ms + tick ; } # [test_only] public fun set_for_testing (clock : & mut Clock , timestamp_ms : u64) { assert ! (timestamp_ms >= clock . timestamp_ms) ; clock . timestamp_ms = timestamp_ms ; } # [test_only] public fun destroy_for_testing (clock : Clock) { let Clock { id , timestamp_ms : _ } = clock ; id . delete () ; }
